contrib/quicktag.lua

```diff
@@ -0,0 +1,220 @@
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+]]
+
+--[[
+QUICKTAG
+   For faster attaching your favorite tags, the script adds shortcuts and
+   the module "quicktag" in lighttable mode with a changeable number of buttons.
+   To each button a tag can be assigned. If the tags do not exist in your database,
+   they are added to the database once they get the first time attached to an image.
+
+   The number of buttons/shortcuts can be changed in the lua preferences.
+   Changes in the number require a restart of darktable.
+
+INSTALATION
+   * copy this file in $CONFIGDIR/lua/ where CONFIGDIR is your darktable configuration directory
+   * add the following line in the file $CONFIGDIR/luarc require "quicktag"
+
+USAGE
+   * set the number of quicktags between 2 and 10 in the preferences dialog and restart darktable
+   * if wanted set the shortcuts in the preferences dialog
+   * to add or change a quicktag, first select the old tag with the combobox "old quicktag",
+     enter a new tag in the "new quicktag" filed and press "set quicktag"
+   * use a shortcut or button to attach the tag to selected images
+
+TODO
+    * abbrevate button labels
+  ]]
+
+
+local dt = require "darktable"
+local debug = require "darktable.debug"
+
+
+local gettext = dt.gettext
+
+dt.configuration.check_version(...,{3,0,0},{4,0,0},{5,0,0})
+
+-- Tell gettext where to find the .mo file translating messages for a particular domain
+gettext.bindtextdomain("quicktag",dt.configuration.config_dir.."/lua/")
+
+local function _(msgid)
+    return gettext.dgettext("quicktag", msgid)
+end
+
+
+
+-- register number of quicktags
+dt.preferences.register("quickTag",         
+                        "quickTagnumber",  
+                        "integer",                    -- type
+                        _("number of quicktag fields"),           -- label
+                        _("may range from 2 to 20 - needs a restart"),   -- tooltip
+                        5,                            -- default
+                        2,                            -- min
+                        20)
+
+local qnr = dt.preferences.read("quickTag", "quickTagnumber", "integer")
+
+-- get quicktags from the preferences
+local quicktag_table = {}
+
+local function read_pref_tags()
+  for j=1,qnr do
+    quicktag_table[j] = dt.preferences.read("quickTag", "quicktag"..j, "string")
+    end
+  end
+
+
+
+-- quicktag function to attach tags
+local function tagattach(tag,qtagnr)
+  if tag == "" then
+    dt.print(_("quicktag").." "..qtagnr.." ".. _("empty, please set a tag"))
```

Comment:
For strings, that needs translations it is better to use string.format(), because in some languages the word order is different.
You can find here an example:
https://github.com/darktable-org/lua-scripts/blob/master/contrib/kml_export.lua#L77